channeldb/payments.go

```diff
@@ -530,6 +553,20 @@ func deserializeHop(r io.Reader) (*route.Hop, error) {
 		return nil, err
 	}
 
+	// TODO(roasbeef): change field to allow LegacyPayload false to be the
+	// legacy default?
```

Comment:
It refers to possibly modifying the on-disk representation to make the modern the default in the control flow. See how it's read/handled in the `serializeHop` method.

Reply:
It sounds like a simple remark, but I still don't get it. "allow LegacyPayload false to be the legacy default". Do you mean to write a 0 in the database for legacy instead of the 1 that is currently written?

Reply:
It just refers to modifying the control flow (or even the struct) to make the legacy the non-default.